src/radical/pilot/agent/rm/base.py

```diff
@@ -74,14 +74,15 @@ def __init__(self, cfg, session):
 
         self._log.info("Configuring LRMS %s.", self.name)
 
-        self.lm_info         = dict()
-        self.lrms_info       = dict()
-        self.slot_list       = list()
-        self.node_list       = list()
-        self.agent_nodes     = dict()
-        self.cores_per_node  = None
-        self.gpus_per_node   = None
-        self.lfs_per_node    = None
+        self.lm_info            = dict()
+        self.lrms_info          = dict()
+        self.slot_list          = list()
+        self.node_list          = list()
+        self.agent_nodes        = dict()
+        self.sockets_per_node   = None
+        self.cores_per_socket   = None
+        self.gpus_per_socket    = None
```

Comment:
The notion of `gpus_per_socket` is not correct. In addition, you removed `cores_per_node` and added `socket_per_node` and `cores_per_socket`.  I suggest that instead of all these changes to go to the base class, they can easily go to the actual LSF resource manager and avoid any huge changes in all resource managers.

As we have discussed, there are GPUs which happen to be closer to a socket compared to the other. Still for the base class it should be as it was and any changes like that should be part of the LSF resource manager class

Reply:
I think we did agree on 'gpus_per_socket', 'cores_per_socket' and 'sockets_per_node' to be used. I agree that these shouldn't be in the base class though. Similar to how we have ContinuousSummit scheduler, I can create rm/lsf_summit which will overwrite all/most methods of base (including the constructor). I have to overwrite the constructor as well as since the operations inside depend on whether we have cores_per_node or cores_per_socket+sockets_per_node. 

I wanted to discuss this in today's call. But I'll leave a comment here for now. I think eventually base class will change as we try to port the socket knowledge to all resource managers (similar to the case of the scheduler). Thanks for looking into it.